src/App.tsx

```diff
@@ -1,26 +1,113 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from 'react';
+import { TodoForma } from './components/TodoForma';
+import { TodoList } from './components/TodoList';
+import { IMessage_Date, ITodo } from './interfaces';
+import { MessageInput } from './components/MessageInput';
+import { MessageOutput } from './components/MessageOutput';
+import { ContextMenu } from './components/ContextMenu';
+
+const App: React.FC = () => {
+
+  const [todos, setTodos] = useState<ITodo[]>([]);
+
+  let chatName_exist:boolean = false;
+
+  const addHandler = (title:string) => {
+    const newTodo: ITodo = {
+      title,
+      id: Date.now(),
+      completed: false,
+      container: [],
+      showContext: false,
+      x:"0px",
+      y:"0px",
+
+    };
+    todos.map((todo) => {
```

Comment:
it is strange to use `.map` in this case... you are mapping each element of array into new array (and in this case you just return todos as an element.

So if previously you hand `todos = [{title:1}, {title:2}, ...]`, then now you have:
`todos = [[[[[[], [[[], [], [],[]], [[], [], [],[]], [[], [], [],[]],[]], [[], [], [],[]],[[], [], [],[]]], [], [],[]], [], [],[]], [], [],[]]` ...

Reply:
probably you want to find an element value: 
```
const found = todos.find(todo => todo.title === title);
if(found) {
alert('Chat with same name already exist!!!')
 return;
}
setTodos(prev => [newTodo, ...prev]);
```

I guess you do not need chatName_exist?